MaxText/train.py

```diff
@@ -450,6 +451,13 @@ def train_loop(config, state=None):
 
     with jax.profiler.StepTraceAnnotation("train", step_num=step):
       example_batch = load_next_batch(data_iterator, example_batch, config)
+      if config.max_checkify:
```

Comment:
This really really shouldn't be in train.py -- can we put it in the step function?

We're trying to keep train.py lean and mean.